### ISSUE TITLE
C-c C-c does not break user program

### ISSUE THREAD

**Issue by riclima** [greghendershott/racket-mode#444]
If you run the following `site.rkt` file:

```racket
#lang web-server/insta
--
(define (start request)
(response/xexpr
'(html
(head (title "My Blog"))
(body (h1 "Under construction")))))
```

on the racket REPL or in DrRackeţ, it automatically starts a web server and prints the log to the console, which blocks the REPL. If you press C-c, it stops the web server, and drops to a REPL with `site.rkt` entered. Is it possible to do the same using the racket-mode's REPL? Pressing C-c or C-d didn't seem to work.

**Comment by greghendershott** [comment]
In DrRacket, pressing <kbd>C-c</kbd> does nothing for me. Instead, pressing the Stop button gives me the REPL prompt. Is that what you meant? (I don't mean to be picky, I just want to make sure I understand.)

---

Assuming that's the case:

Quick glance, the DrRacket Stop button seems to do a `break-thread` of a specific Racket thread. This is different from typical Emacs things like sending a SIGINT or SIGKILL to a process. So, if I were to implement something like this in Racket Mode, I'd need to add an equivalent command to its back end. I'm willing to add such a feature after I understand what would be involved.

**Comment by greghendershott** [comment]
So in the Racket Mode REPL, the way to send a break is to use <kbd>C-c C-c</kbd>. That is bound to `comint-interrupt-subjob` -- which sends a SIGINT to the process running in the Racket REPL, which meant it would break your program... [until recently](https://www.greghendershott.com/2020/02/the-big-switcheroo.html).

Now that there can be multiple REPLs, the processes are Emacs network processes, not subjobs. As a result, if you <kbd>C-c C-c</kbd> Emacs will tell you "comint-interrupt-subjob: Process *Racket REPL* is not a subprocess", and your program won't break.

TL;DR I broke this recently, didn't realize that, and will fix it soon. I already have a fix, but, I'd like to sleep on it to make sure I don't do something else dumb.

**Comment by riclima** [comment]
@greghendershott thank you for taking a look at this! I would have offered to a PR but I'm fairly new to both Racket and Elisp, so I'm not much use with these tools just yet.

**Comment by greghendershott** [comment]
@riclima Oh, I wouldn't have expected you to offer a PR for this.

For one thing, it was an omission and loose end, leftover from implementing multiple REPLs, that I really should have realized and handled, in the first place.

Also it shaded off into some things about how much information we know about a REPL session during its "lifetime". So it became a little more than a few lines, to change.

It was something I ought to fix myself, and, I think I had the best chance to fix correctly --- well, I hope, we'll see about that.

Thank you for reporting it!

**Comment by riclima** [comment]
@greghendershott thanks for the great work!
